js/packages/openinference-instrumentation-llama-index/src/instrumentation.ts

```diff
@@ -0,0 +1,162 @@
+  const suppressTracing = isTracingSuppressed(activeContext);
+  const execContext = suppressTracing
+    ? trace.setSpan(context.active(), span)
+    : activeContext;
+  // Drop the span from the context
+  if (suppressTracing) {
+    trace.deleteSpan(activeContext);
+  }
+  return execContext;
+}
+export class LlamaIndexInstrumentation extends InstrumentationBase<
+  typeof llamaindex
+> {
+  constructor(config?: InstrumentationConfig) {
+    super(
+      "@arizeai/openinference-instrumentation-llama-index",
+      VERSION,
+      Object.assign({}, config),
+    );
+  }
+  manuallyInstrument(module: typeof llamaindex) {
+    diag.debug(`Manually instrumenting ${MODULE_NAME}`);
+    this.patch(module);
+  }
+
+  protected init(): InstrumentationModuleDefinition<typeof llamaindex> {
+    const module = new InstrumentationNodeModuleDefinition<typeof llamaindex>(
+      "llamaindex",
+      [">=0.1.0"],
+      this.patch.bind(this),
+      this.unpatch.bind(this),
+    );
+    return module;
+  }
+
+  private patch(moduleExports: typeof llamaindex, moduleVersion?: string) {
+    this._diag.debug(`Applying patch for ${MODULE_NAME}@${moduleVersion}`);
+    if (_isOpenInferencePatched) {
+      return moduleExports;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const instrumentation: LlamaIndexInstrumentation = this;
+
+    type RetrieverQueryEngineQueryType =
+      typeof moduleExports.RetrieverQueryEngine.prototype.query;
+
+    this._wrap(
+      moduleExports.RetrieverQueryEngine.prototype,
+      "query",
+      (original: RetrieverQueryEngineQueryType): any => {
+        return function patchedQuery(
+          this: unknown,
+          ...args: Parameters<RetrieverQueryEngineQueryType>
+        ) {
+          const span = instrumentation.tracer.startSpan(`Query`, {
+            kind: SpanKind.INTERNAL,
+            attributes: {
+              [SemanticConventions.OPENINFERENCE_SPAN_KIND]:
+                OpenInferenceSpanKind.CHAIN,
+              [SemanticConventions.INPUT_VALUE]: args[0].query,
+            },
+          });
+
+          const execContext = getExecContext(span);
+
+          const execPromise = safeExecuteInTheMiddle<
+            ReturnType<RetrieverQueryEngineQueryType>
+          >(
+            () => {
+              return context.with(execContext, () => {
+                return original.apply(this, args);
+              });
+            },
+            (error) => {
+              // Push the error to the span
+              if (error) {
+                span.recordException(error);
+                span.setStatus({
+                  code: SpanStatusCode.ERROR,
+                  message: error.message,
+                });
+                span.end();
+              }
+            },
+          );
+          const wrappedPromise = execPromise.then((result) => {
+            span.setAttributes({
+              [SemanticConventions.OUTPUT_VALUE]: JSON.stringify(
```

Comment:
assuming vector store stuff is coming in later pr's

Reply:
Yup, just getting off the ground here.